classes/class6/README.md

```diff
@@ -1 +1,96 @@
-#Class 6
+#Class 6: APIs and Debugging Strategies
+##APIs: Your *real* on-ramp to the information superhighway
+
+![Take it to the moon, take it to the stars.](/images/tothemoon.jpg)
+
+*Get in loser, we're going to programmatically acquire JSON data.*
+
+So we've spent some time making our own web services that deliver static and dynamic page content, and we've done it all with the semantics of REST (GET, POST, PUT, DELETE). Today we see how we can use REST APIs to deliver content other than webpages (data!), and how we can access existing APIs set up by some of our favorite websites. Let's get started.
+
+##What's an API?
+Generally the acronym API (Application Programming Interface) is used to describe the way in which one piece of software exposes some of its functionality for another piece of software to use. When we talk about APIs in this class we're going to be specifically referring to web APIs, which most commonly expose their functionality through a specific URL or set of URLs. The APIs we'll be working with will allow you to programmatically (read: without a browser) GET and POST data from 3rd party web services.
+
+##Some Examples
+There are a lot of APIs out there that give you access to more types of data than you'll ever know what to do with. You can check out a long but by no means exhaustive list [here](http://www.programmableweb.com/category/all/apis?data_format=21190). That's a lot of data sitting there for the taking!
+
+###Twitter
+Search for tweets and users, post statuses, read entire timelines (all tweets from a user's following list), and more. Useful for gathering large amounts of data for natural language processing, creating a better interface for twitter, analytics platforms, or tweeting bots! **Examples:** [Tweetbot Twitter client](http://tapbots.com/software/tweetbot/), [Tospy Analytics](http://topsy.com/analytics), [Pixelsorter Bot](https://twitter.com/pixelsorter)
+
+>**URL Endpoint:** https://api.twitter.com/1.1
+
+>**Data Format**: JSON
+
+>**Account Creation:** Automatic key assignment
+
+>**Auth Scheme:** OAuth 1.0a
+
+>**Documentation:** https://dev.twitter.com/rest/public
+
+###Last.fm
+Get information on music artist, albums, and individual tracks. Get listening trend data, even venue and event info. You can also use this API to record users' track listens (called scrobbling) to Last.fm. **Examples:** [Tastebuds Music-based Dating](http://tastebuds.fm/), [Last.fm Extra Stats](http://www.last.fm/user/C26000/journal/2006/07/30/383m_last.fm_extra_stats), [Spotibot Playlist Generator](http://www.spotibot.com/)
+
+>**URL Endpoint:** http://ws.audioscrobbler.com/2.0/
+
+>**Data Format**: XML/JSON if requested
```

Comment:
Would a small mention of XML formatting be appropriate (or a link to a reference)?